openwisp_users/models.py

```diff
@@ -59,6 +60,14 @@ def organizations_pk(self):
                     .values_list('organization_id')
         return qs
 
+    def clean(self):
+        if self.email:
+            try:
+                User.objects.exclude(pk=self.pk).get(email=self.email)
+                raise ValidationError({'email': 'A user with this email already exist'})
+            except User.DoesNotExist:
+                pass
+
```

Comment:
could you also add a new index to make sure that query runs fast?

```
index_together = (
    ('pk', 'email')
)
```

Read more about https://docs.djangoproject.com/en/2.2/ref/models/options/#index-together